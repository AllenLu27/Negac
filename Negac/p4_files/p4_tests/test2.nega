void fn_while(){
	int z;
	z = 10;
     while(z < 10){
	z--;
	}
}

void fn_while2(){
       bool a;
       bool b;
       int c;
       c = 20;
       a = false;
       b = false;
       while(a){
	while(b){
       	  if(b){
		c--;
	       }
       }
    }
}
