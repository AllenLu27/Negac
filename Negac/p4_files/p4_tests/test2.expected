[BEGIN GLOBALS]
[END GLOBALS]
[BEGIN fn_while LOCALS]
z (local)
tmp0 (tmp)
[END fn_while LOCALS]
enter fn_while
z := 10
lbl_1: nop
tmp0 := z LT 10
iffalse tmp0 goto lbl_2
z := z SUB 1
goto lbl_1
lbl_2: nop
lbl_0: leave fn_while
[BEGIN fn_while2 LOCALS]
a (local)
b (local)
c (local)
[END fn_while2 LOCALS]
enter fn_while2
c := 20
a := 0
b := 0
lbl_4: nop
iffalse a goto lbl_5
lbl_6: nop
iffalse b goto lbl_7
iffalse b goto lbl_8
c := c SUB 1
lbl_8: nop
goto lbl_6
lbl_7: nop
goto lbl_4
lbl_5: nop
lbl_3: leave fn_while2
